### DISCUSSION TITLE
Discarding original after upload conversion

### DISCUSSION THREAD

**Discussion by studiomav** [rr-/szurubooru#440] [Q&A] [answered]
I modified the generate_alternate_formats() function so I could automatically convert formats besides GIF, in my case webm -> mp4 for broader mobile device support.
It works great except I'm left with two associated posts, all I want is the converted one.
I poked about in post_api.py but couldn't figure out how the whole sessions thing works, I'm not very experienced with python.
Can someone guide me in the right direction? Thanks.

**Comment by sgsunder** [comment, marked as answer]
If you want to do it natively:
Try editing `update_post_content()` in `func.posts` to include something like this:
```python3
if post.mime_type == "video/webm":
    content = images.Image(content).to_mp4()
```
which I think should do the trick.

An alternate, more robust way of doing this is to use the webhooks feature. I do something similar for cleaning up tags on my personal instance. You can create a lightweight flask (or whatever framework you prefer) container in Docker that's hosted in parallel to the main szurubooru instance.

When a new post is made, a webhook call will be made to this server, which can then perform the webm to mp4 conversion, which can then make a PUT request and update the content of the post.

**Reply by studiomav** [reply]
Thanks for the suggestion, and I believe you're on the right track. I reverted all my changes then updated `update_post_content()` with your addition. Uploading a webm takes some extra time so I do believe it is going through the conversion process (It also took the same extra time with my own modifications that was double-posting both formats). However, the completed post is still webm format. For whatever reason, it's just not "sticking".

```
    elif mime.is_video(post.mime_type):
        post.type = model.Post.TYPE_VIDEO
        if post.mime_type == "video/webm":
            content = images.Image(content).to_mp4()
```

**Reply by studiomav** [reply]
Aha, I got it working!
In `create_post()` I changed `post.mime_type = ""` to 
```
post.mime_type = mime.get_mime_type(content)
if post.mime_type == "video/webm":
     content = images.Image(content).to_mp4()
```

Then `update_post_content(post, content)` is called immediately after. I guess the conversion needed to take place before assessing the metadata or something like that. Thank you @sgsunder for the assistance!
